indico/modules/events/surveys/controllers/display.py

```diff
@@ -112,13 +121,30 @@ def _process(self):
                                                event=self.event_new, survey=self.survey,
                                                back_button_endpoint=back_button_endpoint)
 
-    def _save_answers(self, form):
+    def _make_form(self):
+        survey_form_class = make_survey_form(self.survey)
+        if self.submission and request.method == 'GET':
+            return survey_form_class(formdata=MultiDict(self.submission.pending_answers))
+        else:
+            return survey_form_class()
+
+    @no_autoflush
+    def _save_answers(self, form, submission):
         survey = self.survey
-        submission = SurveySubmission(survey=survey)
-        if not survey.anonymous:
-            submission.user = session.user
+        if not submission:
+            submission = SurveySubmission(survey=survey, user=session.user)
+        submission.is_anonymous = survey.anonymous
```

Comment:
I don't get your point. It's already in `_save_answers`. Do you mean moving it to `_process()` below `_save_answers` is called or create a new function `save_answer()`?